src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/KeyworkerService.java

```diff
@@ -135,25 +135,20 @@ private static String getKeyWorkerFullName(final KeyworkerDto keyworkerDto) {
         return ConversionHelper.convertOffenderKeyworkerModel2Dto(results);
     }
 
-    public Optional<BasicKeyworkerDto> getCurrentKeyworkerForPrisoner(final String prisonId, final String offenderNo) {
-         BasicKeyworkerDto currentKeyworker = null;
-        if (prisonSupportedService.isMigrated(prisonId)) {
-            final var activeOffenderKeyworker = repository.findByOffenderNoAndActiveAndAllocationTypeIsNot(offenderNo, true, AllocationType.PROVISIONAL);
-            if (activeOffenderKeyworker != null) {
-                final var staffDetail = nomisService.getBasicKeyworkerDtoForStaffId(activeOffenderKeyworker.getStaffId());
-                if (staffDetail != null) {
-                    currentKeyworker = BasicKeyworkerDto.builder()
-                            .firstName(staffDetail.getFirstName())
-                            .lastName(staffDetail.getLastName())
-                            .staffId(staffDetail.getStaffId())
-                            .email(staffDetail.getEmail())
-                            .build();
-                }
+    public Optional<BasicKeyworkerDto> getCurrentKeyworkerForPrisoner(final String offenderNo) {
+        final var activeOffenderKeyworker = repository.findByOffenderNoAndActiveAndAllocationTypeIsNot(offenderNo, true, AllocationType.PROVISIONAL);
```

Comment:
change here - always go to the repository to find the keyworker, even if the prison not migrated